sequoia/methods/d3rlpy_methods/base_test.py

```diff
@@ -0,0 +1,124 @@
+import pytest
+from d3rlpy.constants import ActionSpace
+
+from sequoia.methods.d3rlpy_methods.base import *
+from sequoia.settings.offline_rl.setting import OfflineRLSetting
+
+
+class BaseOfflineRLMethodTests:
+    Method: ClassVar[Type[BaseOfflineRLMethod]]
+
+    @pytest.fixture
+    def method(self):
+        return self.Method(train_steps=1, train_steps_per_epoch=1)
+
+    @pytest.mark.parametrize('dataset', OfflineRLSetting.available_datasets)
+    def test_offlinerl(self, method, dataset: str):
+
+        setting_offline = OfflineRLSetting(dataset=dataset)
+
+        #
+        # Check for mismatch
+        if isinstance(setting_offline.env.action_space, gym.spaces.Box):
+            if method.algo.get_action_type() is not ActionSpace.CONTINUOUS:
+                return
+
+        elif isinstance(setting_offline.env.action_space, gym.spaces.discrete.Discrete):
+            if method.algo.get_action_type() is not ActionSpace.DISCRETE:
+                return
+        else:
+            return
+
+        results = setting_offline.apply(method)
+
+        # Check that the metric dict for our 1 step epoch is not None
+
+        epoch_metrics = results[-1][1]
+        assert epoch_metrics is not None
+        assert isinstance(epoch_metrics, dict)
+
+    @pytest.mark.parametrize('dataset', TraditionalRLSetting.available_datasets)
+    def test_traditionalrl(self, method, dataset):
+
+        # BC is a strictly offline method
+        if type(method) in {BCMethod, BCQMethod, DiscreteBCMethod, DiscreteBCQMethod}:
```

Comment:
Ok there's something interesting here. I'll need more time to explain it in a meeting though.

Your offline RL methods all have the OfflineRLSetting as their target setting. `<your method class>is_applicable()` is the method that gets called to check if a method can be used on a given setting. 
Therefore, they currently are all saying that they aren't applicable on anything that isn't an `OfflineRLSetting`. (and their `is_applicable` method returns `False` for the TraditionalRLSetting, for example).

For now, methods are limited to one target setting. This will change in the future.
Once I allow specifying multiple target settings or passing a list of assumptions (sometime after december 1st), then we can update the methods that work in both, so that they target both the `RLSetting` and `OfflineRLSetting`. This will make them available from the command-line also for those settings.


One other small thing: use `pytest.skip("Method only works on offline RL settings")` rather than return.

Reply:
I added pytest.skip() with specific information in my tests instead of returning. Lets chat about the rest on Friday

Reply:
Oh also: use `isinstance(method, (BCMethod, BCQMethod, DiscreteBCMethod, DiscreteBCQMethod))` instead of `type(method) in`

Reply:
switched to isinstance